### DISCUSSION TITLE
NiceGUI docker image with docker out of docker (DooD)

### DISCUSSION THREAD

**Discussion by J0ergL** [zauberzeug/nicegui#3730] [Q&A]
### Question

I am trying to setup docker out of docker. 
Why you might ask. 
I want to restart another container running on the same host as the niceGUI container via a niceGUI button.

my dockerfile looks like this:
```
FROM zauberzeug/nicegui
EXPOSE 8080
RUN apt-get update && \
    DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC \
    apt-get install -y --no-install-recommends \
    [docker.io](http://docker.io/) \
    && rm -rf /var/lib/apt/lists/* \
    && apt-get clean

RUN groupadd -r appgroup && useradd -r -g appgroup appuser
RUN usermod -aG docker appuser
ARG HOST_DOCKER_GID=111 # this is raspian bookwork specific, aquired with "getent group docker | cut -d: -f3"
RUN groupmod -g $HOST_DOCKER_GID docker
CMD ["python3", "jkastenGUI.py"]
```
building works
`docker build -t nicegui-docker .`

but I cannot start it:
`docker run --rm -itv /var/run/docker.sock:/var/run/docker.sock nicegui-docker bash`

returns: 
```
groupadd: group 'appgroup' already exists
useradd: group '1000' does not exist
cp: cannot create regular file '/home/appuser/.bashrc': No such file or directory
chown: cannot access '/home/appuser/.bashrc': No such file or directory
setpriv: uid 1000 not found, --init-groups requires an user that can be found on the system
```
I am looking forward to your suggestions on how to accomplish DooD with NiceGUI.

**Comment by rodja** [comment]
Our zauberzeug/nicegui image uses [an entry point to create group and user](https://github.com/zauberzeug/nicegui/blob/40e9f094bfe9c0940d7e7a76cce49dd7e446f5c9/release.dockerfile#L37-L43). If you create them already in your Dockerfile the entry point fails. You could provide your own "docker-entrypoint.sh" via COPY which does not create the appuser and appgroup.
